torch/csrc/autograd/variable.cpp

```diff
@@ -127,22 +127,24 @@ std::shared_ptr<Function>& VariableViewImpl::get_grad_fn() {
   return _grad_fn;
 }
 
-void VariableViewImpl::rebase_grad_fn(std::shared_ptr<Function> grad_fn) {
+void VariableViewImpl::rebase_history(VarFlags flags, int output_nr, std::shared_ptr<Function> grad_fn) {
   TORCH_ASSERT(output_nr == 0);
-  if (!grad_fn) {
-    TORCH_ASSERTM(!requires_grad, "Can't set null grad_fn on view with requires_grad=True");
+  TORCH_ASSERT(flags.requires_grad == bool(grad_fn));
+  if (grad_fn) {
+    TORCH_ASSERTM(grad_fn->num_inputs == 1, "Functions which modify views in-place must return a single Variable");
```

Comment:
We should really have `TORCH_ASSERT` for invariants and `TORCH_EXPECT` for checks that might fail as a result of user interactions